evalml/tests/component_tests/detrender_tests/test_polynomial_decomposer.py

```diff
@@ -0,0 +1,203 @@
+
+    with pytest.raises(TypeError, match="Received float"):
+        PolynomialDecomposer(degree=3.4)
+
+    _ = PolynomialDecomposer(degree=3.0)
+
+
+def test_polynomial_decomposer_raises_value_error_target_is_none(ts_data):
+    X, y = ts_data
+
+    with pytest.raises(ValueError, match="y cannot be None for PolynomialDecomposer!"):
+        PolynomialDecomposer(degree=3).fit_transform(X, None)
+
+    with pytest.raises(ValueError, match="y cannot be None for PolynomialDecomposer!"):
+        PolynomialDecomposer(degree=3).fit(X, None)
+
+    pdt = PolynomialDecomposer(degree=3).fit(X, y)
+
+    with pytest.raises(ValueError, match="y cannot be None for PolynomialDecomposer!"):
+        pdt.inverse_transform(None)
+
+
+def test_polynomial_decomposer_raises_value_error_transform_fit_data_freq_mismatch(
+    ts_data,
+):
+    X, y = ts_data
+
+    pdc = PolynomialDecomposer(degree=3).fit(X, y)
+
+    with pytest.raises(ValueError, match="Cannot transform given data with frequency"):
+        # Change target data frequency to monthly to mismatch daily frequency of fixture
+        y = y.asfreq("m")
+        pdc.transform(X, y)
+
+
+def test_polynomial_decomposer_get_trend_df_raises_errors(ts_data):
+    X, y = ts_data
+    pdt = PolynomialDecomposer(degree=3)
+    pdt.fit_transform(X, y)
+
+    with pytest.raises(
+        TypeError, match="Provided X should have datetimes in the index."
+    ):
+        X_int_index = X.reset_index()
+        pdt.get_trend_dataframe(X_int_index, y)
+
+    with pytest.raises(TypeError, match="y must be pd.Series or pd.DataFrame!"):
+        y = np.array(y.values)
+        pdt.get_trend_dataframe(X, y)
+
+    with pytest.raises(
+        ValueError,
+        match="Provided DatetimeIndex of X should have an inferred frequency.",
+    ):
+        X.index.freq = None
+        pdt.get_trend_dataframe(X, y)
+
+
+@pytest.mark.parametrize("input_type", ["pd", "ww"])
+@pytest.mark.parametrize("degree", [1, 2, 3])
+def test_polynomial_decomposer_fit_transform(degree, input_type, ts_data):
+
+    X_input, y_input = ts_data
+
+    # Get the expected answer
+    lin_reg = LinearRegression(fit_intercept=True)
+    features = PolynomialFeatures(degree=degree).fit_transform(
+        np.arange(X_input.shape[0]).reshape(-1, 1)
+    )
+    lin_reg.fit(features, y_input)
+    detrended_values = y_input.values - lin_reg.predict(features)
+    expected_index = y_input.index if input_type != "np" else range(y_input.shape[0])
+    expected_answer = pd.Series(detrended_values, index=expected_index)
+
+    X, y = X_input, y_input
+
+    if input_type == "ww":
+        X = X_input.copy()
+        X.ww.init()
+        y = ww.init_series(y_input.copy())
+
+    output_X, output_y = PolynomialDecomposer(degree=degree).fit_transform(X, y)
+    pd.testing.assert_series_equal(expected_answer, output_y)
+
+    # Verify the X is not changed
+    pd.testing.assert_frame_equal(X, output_X)
+
+
+@pytest.mark.parametrize(
```

Comment:
This is a new test and I added support to just this function for multivariate targets.